pvmismatch/pvmismatch_lib/pvcell.py

```diff
@@ -127,11 +183,14 @@ def Aph(self):
         if self.Isc == 0: return np.nan
         # short current (SC) conditions (Vcell = 0)
         Vdiode_sc = self.Isc * self.Rs  # diode voltage at SC
-        Idiode1_sc = self.Isat1 * (np.exp(Vdiode_sc / self.Vt) - 1.)
-        Idiode2_sc = self.Isat2 * (np.exp(Vdiode_sc / 2. / self.Vt) - 1.)
+        Idiode1_sc = self.Isat1 * (np.expm1(Vdiode_sc / self.N1 / self.Vt))
+        Idiode2_sc = self.Isat2 * (np.expm1(Vdiode_sc / self.N2 / self.Vt))
         Ishunt_sc = Vdiode_sc / self.Rsh  # diode voltage at SC
         # photogenerated current coefficient
-        return 1. + (Idiode1_sc + Idiode2_sc + Ishunt_sc) / self.Isc
+        if self.diode_model=='2diode':
+            return 1. + (Idiode1_sc + Idiode2_sc + Ishunt_sc) / self.Isc
```

Comment:
Ah, re: [my comment above](#r387474969) I see now, it's fine to leave N2 as non-zero, because it won't be included in the desoto or pvsyst calcs anyway. Clever!